python/pyarrow/tests/test_convert_pandas.py

```diff
@@ -160,9 +160,28 @@ def test_integer_index_column(self):
         df = pd.DataFrame([(1, 'a'), (2, 'b'), (3, 'c')])
         _check_pandas_roundtrip(df, preserve_index=True)
 
+    def test_index_metadata_field_name(self):
+        df = pd.DataFrame(
+            [(1, 'a'), (2, 'b'), (3, 'c')],
+            index=pd.MultiIndex.from_arrays(
+                [['c', 'b', 'a'], [3, 2, 1]],
+                names=[None, 'foo']
+            )
+        )
+        t = pa.Table.from_pandas(df, preserve_index=True)
+        raw_metadata = t.schema.metadata
+
+        js = json.loads(raw_metadata[b'pandas'].decode('utf8'))
+
+        _, _, idx0, foo = js['columns']
+        idx0_name, foo_name = js['index_columns']
+        assert idx0_name == '__index_level_0__'
+        assert idx0['field_name'] == idx0_name
+
+        assert foo_name == '__index_level_1__'
+        assert foo['name'] == 'foo'
+
```

Comment:
For completeness, I would assert that for the other columns, the name and field_name are equal (this is obvious from the current code, but not explicitly tested):

```
col1, col2, idx0, foo = js['columns']

assert col1['name'] == col1['field_name']
assert col2['name'] == col2['field_name']
```

Reply:
Yep, will do